pkgs/tools/networking/smartdns/default.nix

```diff
@@ -0,0 +1,54 @@
+
+{ stdenv, fetchFromGitHub, openssl, fetchpatch }:
+
+stdenv.mkDerivation rec {
+  pname = "smartdns";
+  version = "29";
+  src = fetchFromGitHub {
+    owner = "pymumu";
+    repo = pname;
+    rev = "Release${version}";
+    sha256 = "06kgmahgma0ybhacqzj5sfazmhzs3p27a0sxik4xbbk3dyqn9ykj";
+  };
+
+  buildInputs = [ openssl ];
+
+  patches = [
+    (fetchpatch {
+      url =
+        "https://git.archlinux.org/svntogit/community.git/plain/trunk/systemd.patch?h=packages/smartdns";
+      sha256 = "0jxzsgxz6y1a4mvwk6d5carxv4nxsswpfs1a6jkplzy2gr0g8281";
+    })
+  ];
+
+  prePatch = "chmod -R +w ../systemd";
+
+  patchFlags = [ "-p1" "../systemd/smartdns.service" ];
```

Comment:
This will be problematic when multiple patches are added. Perhaps remove the `sourceRoot` invocation – that should allow you to get rid of the `prePatch` as well.

You will need to add `makeFlags = [ "-C" "src" ];` or `preBuild = ''cd src'';` instead.

Reply:
Got it